pyfixest/estimation/vcov_utils.py

```diff
@@ -96,6 +96,88 @@ def _get_vcov_type(
 
     return vcov_type  # type: ignore
 
+@nb.njit(parallel=False)
+def _nw_meat(scores, time_var=None, lags=None, data=None, is_iv=False, 
+             tXZ=None, tZZinv=None, tZX=None):
+    """
+    Compute Newey-West HAC meat matrix.
+    Parameters
+    ----------
+    scores: np.ndarray
+        The scores matrix.
+    time_var: np.ndarray, optional
+        The time variable for clustering. Default is None.
+    lags: int, optional
+        The number of lags for the HAC estimator. Defaults to floor (# of time periods)^(1/4).
+    data: pd.DataFrame, optional
+        The original data set. Default is None.
+    is_iv: bool, optional
+        Whether the model is an instrumental variable model. Default is False.
+    tXZ: np.ndarray, optional
+        The t(X)'Z matrix for IV models. Default is None.   
+    
+    tZZinv: np.ndarray, optional
+        The (Z'Z)^(-1) matrix for IV models. Default is None.
+    tZX: np.ndarray, optional
+        The t(Z)X matrix for IV models. Default is None.
+    """
+
+    # Determine whether data is time ordered or not - if not provided, data is assumed to be time ordered
+    if time_var is None:
+        ordered_scores = scores
+        n_time = len(ordered_scores)
+    else:
+        time_data = data[time_var].to_numpy() # need to ensure that this is datetime format - might have to write an internal converter for pf.
+        order = np.argsort(time_data)
+        ordered_scores = scores[order]
+        n_time = len(np.unique(time_data))
+
+    # resolve lags
+    if lags is None:
+        lags = int(np.floor(n_time ** (1 / 4)))
+
+    # bartlett kernel weights
+    weights = np.array([1 - j/(lags + 1) for j in range(lags + 1)])
+    weights[0] = 0.5  # Halve first weight
+
+    n, k = ordered_scores.shape
+    meat = np.zeros((k, k))
+
+    for j in range(lags + 1):
+        if j == 0:
+            gamma_j = ordered_scores.T @ ordered_scores
+
+
+            meat += weights[j] * gamma_j
+        else:
+            gamma_j = ordered_scores[j:].T @ ordered_scores[:-j]
+            meat += weights[j] * (gamma_j + gamma_j.T)
+
+    meat = meat / n
```

Comment:
I don't think you have to divide by /n here, these should cancel out when forming the full vcov matrix.